File: banco_problemas_aplicados/nivel_basico/soluciones/c/problema19.c
/*
 * Problema 19 - Inversion Espejo de Primera Mitad
 * void espejoPrimeraMitad(double Arr[], int n)
 * Contexto: pruebas de impulso. Copiar la primera mitad del vector reflejada en la segunda
 * mitad, dejando el arreglo simetrico:  Arr[n-1-i] = Arr[i].
 */
#include <stdio.h>

void espejoPrimeraMitad(double arr[], int n) {
    for (int i = 0; i < n / 2; i++) {
        arr[n - 1 - i] = arr[i];     /* la posicion espejo copia el valor de la primera mitad */
    }
}

int main(void) {
    int n;
    printf("Ingrese el tamano del vector: ");
    scanf("%d", &n);

    double arr[n];
    for (int i = 0; i < n; i++) {
        printf("Dato %d: ", i);
        scanf("%lf", &arr[i]);
    }

    espejoPrimeraMitad(arr, n);

    printf("Vector simetrizado: ");
    for (int i = 0; i < n; i++) {
        printf("%.2f ", arr[i]);
    }
    printf("\n");
    return 0;
}

File: banco_problemas_aplicados/nivel_basico/soluciones/java/Problema19.java
/*
 * Problema 19 - Inversion Espejo de Primera Mitad
 * void espejoPrimeraMitad(double[] arr, int n)  ->  arr[n-1-i] = arr[i]  (queda simetrico)
 */
import java.util.Scanner;
import java.util.Locale;

public class Problema19 {

    static void espejoPrimeraMitad(double[] arr, int n) {
        for (int i = 0; i < n / 2; i++) {
            arr[n - 1 - i] = arr[i];
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese el tamano del vector: ");
        int n = sc.nextInt();

        double[] arr = new double[n];
        for (int i = 0; i < n; i++) {
            System.out.print("Dato " + i + ": ");
            arr[i] = sc.nextDouble();
        }

        espejoPrimeraMitad(arr, n);

        System.out.print("Vector simetrizado: ");
        for (int i = 0; i < n; i++) {
            System.out.printf(Locale.US, "%.2f ", arr[i]);
        }
        System.out.println();
        sc.close();
    }
}
